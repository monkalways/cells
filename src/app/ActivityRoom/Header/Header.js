import React from 'react';
import PropTypes from 'prop-types';
import {
  AppBar, IconButton, Toolbar, withStyles,
} from '@material-ui/core';
import CloseIcon from '@material-ui/icons/Close';

import HeaderContent from './HeaderContent';
import logoIcon from '../../images/eps-logo-icon.png';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  usage: PropTypes.string.isRequired,
  onLogout: PropTypes.func.isRequired,
  isAuthenticated: PropTypes.bool.isRequired,
};

const Header = ({
  classes, usage, onLogout, isAuthenticated,
}) => (
  <div className={classes.root}>
    <AppBar position="static" className={classes.appBar}>
      <Toolbar>
        <IconButton
          className={classes.menuButton}
          color="inherit"
          aria-label="Menu"
        >
          <img src={logoIcon} alt="EPS Logo" width={70} />
        </IconButton>
        <HeaderContent usage={usage} />
        {isAuthenticated && (
          <IconButton
            className={classes.logoutButton}
            color="inherit"
            aria-label="Menu"
            onClick={onLogout}
          >
            <CloseIcon className={classes.logoutIcon} />
          </IconButton>
        )}
      </Toolbar>
    </AppBar>
  </div>
);

Header.propTypes = propTypes;

export default withStyles((theme) => ({
  root: {
    flexGrow: 1,
  },
  appBar: {
    backgroundColor: theme.palette.background.default,
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20,
  },
  logoutButton: {
    margin: theme.spacing.unit,
    backgroundColor: '#444480',
    color: '#F10607',
    '&:hover': {
      backgroundColor: '#444480',
    },
  },
  logoutIcon: {
    fontSize: theme.typography.h4.fontSize,
  },
}))(Header);
