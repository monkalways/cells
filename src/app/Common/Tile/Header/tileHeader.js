import React from 'react';
import { PropTypes } from 'prop-types';
import CardHeader from '@material-ui/core/CardHeader';
import DetaineeLocationLogo from '../../DetaineeLocationLogo';

const styles = {
  main: {
    padding: '0 3px',
  },
  title: {
    fontSize: 16,
    fontWeight: 600,
    lineHeight: 0.15,
    paddingTop: 0,
  },
  subheader: {
    fontSize: 14,
    fontWeight: 600,
  },
  locationLogo: {
    height: 50,
    width: 50,
    top: 10,
    right: 20,
  },
};

const TileHeader = (props) => {
  const { detainee } = props;
  const strDivisions = detainee.division.join(', ').trim();

  return (
    <CardHeader
      style={styles.main}
      title={(
        <div style={styles.title}>
          <p>{detainee.firstName}</p>
          <p>{detainee.lastName}</p>
          <p style={styles.subheader}>{strDivisions}</p>
        </div>
)}
      action={(
        <DetaineeLocationLogo
          location={detainee.location}
          style={styles.locationLogo}
        />
)}
    />
  );
};

TileHeader.propTypes = {
  detainee: PropTypes.shape({
    id: PropTypes.string.isRequired,
    arrestId: PropTypes.string.isRequired,
    firstName: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired,
    division: PropTypes.arrayOf(PropTypes.string).isRequired,
    detentionUnitName: PropTypes.string,
    location: PropTypes.string.isRequired,
    intakePhotoResourceUri: PropTypes.string,
    gender: PropTypes.string,
    withCaution: PropTypes.bool,
    cautionsArray: PropTypes.arrayOf(PropTypes.string),
    mustBeKeptAlone: PropTypes.bool,
    isSuicidal: PropTypes.bool,
    isContagious: PropTypes.bool,
    hasWarning: PropTypes.bool,
    isUnderMedication: PropTypes.bool,
  }).isRequired,
};

export default TileHeader;
