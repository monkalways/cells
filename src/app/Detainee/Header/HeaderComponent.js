import React from 'react';
import PropTypes from 'prop-types';
import {
  AppBar,
  Grid,
  IconButton,
  Toolbar,
  withStyles,
} from '@material-ui/core';
import CloseIcon from '@material-ui/icons/Close';

import logoIcon from './eps-logo-icon.png';
import HeaderContent from './HeaderContent';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  detainee: PropTypes.shape({}),
  isDetaineeProfileLoaded: PropTypes.bool.isRequired,
  onLogout: PropTypes.func.isRequired,
};

const defaultProps = {
  detainee: null,
};

const Header = ({
  classes, detainee, isDetaineeProfileLoaded, onLogout,
}) => (
  <div className={classes.root}>
    <AppBar position="static" className={classes.appBar}>
      <Toolbar>
        <IconButton
          className={classes.menuButton}
          color="inherit"
          aria-label="Menu"
        >
          <img src={logoIcon} alt="EPS Logo" width={70} />
        </IconButton>
        {isDetaineeProfileLoaded ? (
          <HeaderContent
            detainee={detainee}
            isDetaineeProfileLoaded={isDetaineeProfileLoaded}
          />
        ) : (
          <Grid item xs={12} />
        )}
        <IconButton
          className={classes.logoutButton}
          color="inherit"
          aria-label="Menu"
          onClick={onLogout}
        >
          <CloseIcon className={classes.logoutIcon} />
        </IconButton>
      </Toolbar>
    </AppBar>
  </div>
);

Header.propTypes = propTypes;
Header.defaultProps = defaultProps;

export default withStyles((theme) => ({
  root: {
    flexGrow: 1,
  },
  appBar: {
    backgroundColor: '#A8C6FA', // TODO: move color to theme
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20,
  },
  logoutButton: {
    margin: theme.spacing.unit,
    backgroundColor: '#444480',
    color: '#F10607',
    '&:hover': {
      backgroundColor: '#444480',
    },
  },
  logoutIcon: {
    fontSize: theme.typography.h4.fontSize,
  },
}))(Header);
