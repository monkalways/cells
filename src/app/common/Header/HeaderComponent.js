import React from 'react';
import PropTypes from 'prop-types';
import {
  AppBar, Toolbar, Typography, withStyles,
} from '@material-ui/core';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  versions: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    version: PropTypes.string.isRequired,
  })).isRequired,
};

export const HeaderComponent = ({ classes, versions }) => (
  <AppBar position="static">
    <Toolbar>
      <Typography
        id="title"
        variant="h6"
        color="inherit"
        className={classes.heading}
      >
        Cell App - Detainee Management Unit
      </Typography>
      <div>
        {versions
          && versions.map((version) => (
            <Typography
              id="version"
              color="inherit"
              align="right"
              key={version.name}
            >
              {version.name}: {version.version}
            </Typography>
          ))}
      </div>
    </Toolbar>
  </AppBar>
);

HeaderComponent.propTypes = propTypes;

export default withStyles(() => ({
  heading: {
    flex: 1,
  },
}))(HeaderComponent);
