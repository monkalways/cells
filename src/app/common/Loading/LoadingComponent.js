import React from 'react';
import PropTypes from 'prop-types';
import { CircularProgress, withStyles } from '@material-ui/core';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  size: PropTypes.number,
};

const defaultProps = {
  size: 100,
};

const LoadingComponent = ({ classes, size }) => (
  <div className={classes.container}>
    <CircularProgress className={classes.progress} size={size} />
  </div>
);

LoadingComponent.propTypes = propTypes;
LoadingComponent.defaultProps = defaultProps;

export default withStyles((theme) => ({
  container: {
    display: 'flex',
    width: '100%',
    height: '100%',
    justifyContent: 'center',
    alignItems: 'center',
  },
  progress: {
    margin: theme.spacing.unit * 2,
  },
}))(LoadingComponent);
