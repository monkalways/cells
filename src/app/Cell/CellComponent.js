import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Typography, withStyles } from '@material-ui/core';
import { withRouter } from 'react-router-dom';
import { compose } from 'recompose';

import Layout from './Layout';
import Header from './Header';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      name: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
  initialize: PropTypes.func.isRequired,
  cellDetails: PropTypes.shape({}),
};

const defaultProps = {
  cellDetails: null,
};

class CellComponent extends Component {
  componentDidMount() {
    const { match, initialize } = this.props;
    const { name } = match.params;
    initialize(name);
  }

  handleLogout = () => {
    const { history, match } = this.props;
    const { name } = match.params;
    history.push(`/cells/${name}`);
  };

  render() {
    const { cellDetails, match } = this.props;
    return (
      <React.Fragment>
        {cellDetails && (
          <Layout>
            <Header cell={cellDetails} onLogout={this.handleLogout} />
            <Typography variant="h4">{match.params.name}</Typography>
          </Layout>
        )}
      </React.Fragment>
    );
  }
}

CellComponent.propTypes = propTypes;
CellComponent.defaultProps = defaultProps;

export default compose(
  withStyles(() => ({
    container: {},
  })),
  withRouter,
)(CellComponent);
