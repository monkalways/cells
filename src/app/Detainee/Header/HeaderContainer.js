import { connect } from 'react-redux';
import { selectors } from '../duck';
import HeaderComponent from './HeaderComponent';

export const mapStateToProps = (
  state,
  detainee = selectors.getDetaineeState(state),
  isDetaineeProfileLoaded = selectors.isDetaineeProfileLoadedState(state),
) => ({
  detainee,
  isDetaineeProfileLoaded,
});

export default connect(
  mapStateToProps,
  null,
)(HeaderComponent);
