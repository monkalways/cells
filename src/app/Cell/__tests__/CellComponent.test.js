import React from 'react';
import { shallow } from 'enzyme';

import { CellComponent } from '../CellComponent';
import Header from '../Header';
import commonConstants from '../../constants';
import Loading from '../../common/Loading';

describe('CellComponent', () => {
  let classes;
  let cellDetails;
  let isCellDetaineesLoaded;
  let isAuthenticated;
  let cellName;
  let getCellDetails;
  let handleLogOut;
  let push;
  let refreshAuthenticationTimeout;
  let startAuthenticationTimeout;
  let stopAuthenticationTimeout;

  beforeEach(() => {
    classes = {
      body: 'body',
    };
    cellDetails = null;
    isAuthenticated = false;
    cellName = 'c1';
    getCellDetails = jest.fn();
    handleLogOut = jest.fn();
    push = jest.fn();
    refreshAuthenticationTimeout = jest.fn();
    startAuthenticationTimeout = jest.fn();
    stopAuthenticationTimeout = jest.fn();

    jest.useFakeTimers();
  });

  afterEach(() => {
    jest.useRealTimers();
  });

  const setup = () => shallow(<CellComponent
    classes={classes}
    history={{ push }}
    match={{
      params: {
        name: cellName,
      },
    }}
    cellDetails={cellDetails}
    isCellDetaineesLoaded={isCellDetaineesLoaded}
    isAuthenticated={isAuthenticated}
    handleLogOut={handleLogOut}
    getCellDetails={getCellDetails}
    refreshAuthenticationTimeout={refreshAuthenticationTimeout}
    startAuthenticationTimeout={startAuthenticationTimeout}
    stopAuthenticationTimeout={stopAuthenticationTimeout}
  />);

  it('should render if cell details is not null', () => {
    cellDetails = {};

    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Header)).toExist();
  });

  it('should render Loading component if cell details is null', () => {
    cellDetails = null;

    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Loading)).toExist();
  });

  it('should get cell details when componentDidMount is called', () => {
    const wrapper = setup();

    wrapper.instance().componentDidMount();

    expect(getCellDetails).toBeCalledWith(cellName);
  });

  it('should not set unauthenticated timer when authenticated', () => {
    isAuthenticated = true;
    const wrapper = setup();

    wrapper.instance().componentDidMount();

    expect(setTimeout).not.toBeCalled();
  });

  it(`should set unauthenticated timer to call handle logout in ${
    commonConstants.UNAUTHENTICATED_TIMEOUT_SECONDS
  } seconds when not authenticated and clear timeout after unmount`, () => {
    isAuthenticated = false;
    const wrapper = setup();

    wrapper.instance().componentDidMount();
    expect(setTimeout).toBeCalled();

    jest.advanceTimersByTime(commonConstants.UNAUTHENTICATED_TIMEOUT_SECONDS * 1000);
    expect(handleLogOut).toBeCalledWith('cells', cellName);

    wrapper.instance().componentWillUnmount();
    expect(clearTimeout).toBeCalled();
  });

  it('should cancel unauthenticated timer after authenticated is set to true after initialization', () => {
    isAuthenticated = false;
    const wrapper = setup();

    wrapper.instance().componentDidMount();
    expect(setTimeout).toBeCalled();

    wrapper.setProps({ isAuthenticated: true });
    expect(clearTimeout).toBeCalled();
  });

  it('should start authentication timeout if componentDidUpdate is called and page is authenticated', () => {
    isAuthenticated = true;
    const wrapper = setup();

    wrapper.instance().componentDidUpdate();
    expect(startAuthenticationTimeout).toBeCalled();
  });

  it('should not start authentication timeout if componentDidUpdate is called and page is not authenticated', () => {
    isAuthenticated = false;
    const wrapper = setup();

    wrapper.instance().componentDidUpdate();
    expect(startAuthenticationTimeout).not.toBeCalled();
  });

  it('should refresh authentication timeout on handleClick if page is authenticated', () => {
    isAuthenticated = true;
    const wrapper = setup();

    wrapper.find('#authenticatedCellComponentRoot').simulate('click');
    expect(refreshAuthenticationTimeout).toBeCalled();
  });

  it('should not refresh authentication timeout on handleClick if page is not authenticated', () => {
    isAuthenticated = false;
    const wrapper = setup();

    wrapper.find('#authenticatedCellComponentRoot').simulate('click');
    expect(refreshAuthenticationTimeout).not.toBeCalled();
  });
});
