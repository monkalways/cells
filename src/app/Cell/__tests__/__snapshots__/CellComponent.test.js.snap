// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CellComponent should render if cell details is not null 1`] = `
<Fragment>
  <WithStyles(LayoutComponent)>
    <WithStyles(HeaderComponent)
      cellDetails={Object {}}
      isAuthenticated={false}
      onLogout={[Function]}
    />
    <div
      className="body"
    >
      <Switch>
        <Route
          component={[Function]}
          exact={true}
        />
        <Route
          component={[Function]}
          exact={true}
          path="undefined/cell-check"
        />
        <Route
          component={[Function]}
          exact={true}
          path="undefined/meal"
        />
        <Route
          component={[Function]}
          exact={true}
          path="undefined/Medication"
        />
      </Switch>
    </div>
  </WithStyles(LayoutComponent)>
</Fragment>
`;
